open-vm-tools/toolbox/toolboxcmd-time.c
=======================================================
lhs: 100644 | d6b7a9851508c0107c5a52aea9eb5e01ac34b3d1
rhs: 100644 | 5c011d662c730fb9b62fc47b755fff777b4ea556
---@@ -1,5 +1,5 @@
 /*********************************************************
- * Copyright (C) 2008-2016 VMware, Inc. All rights reserved.
+ * Copyright (C) 2008-2017 VMware, Inc. All rights reserved.
  *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms of the GNU Lesser General Public License as published
@@ -28,6 +28,7 @@
 #include "vmware/guestrpc/tclodefs.h"
 #include "vmware/guestrpc/timesync.h"
 #include "vmware/tools/i18n.h"
+#include "vmware/tools/utils.h"
 
 
 /*
@@ -73,14 +74,16 @@ TimeSyncGetOldOptions(void)
  *-----------------------------------------------------------------------------
  */
 
-static void
-TimeSyncSet(Bool enable) // IN: status
+static gboolean
+TimeSyncSet(Bool enable,      // IN:
+            char **reply,     // OUT:
+            size_t *replyLen) // OUT:
 {
    gchar *msg = g_strdup_printf("vmx.set_option %s %s %s",
                                 TOOLSOPTION_SYNCTIME,
                                 !enable ? "1" : "0",
                                 enable ? "1" : "0");
-   ToolsCmd_SendRPC(msg, strlen(msg) + 1, NULL, NULL);
+   return ToolsCmd_SendRPC(msg, strlen(msg) + 1, reply, replyLen);
 }
 
 
@@ -92,7 +95,7 @@ TimeSyncSet(Bool enable) // IN: status
  *      Enable time sync.
  *
  * Results:
- *      EXIT_SUCCESS
+ *      EXIT_SUCCESS on success, EXIT_FAILURE on failure.
  *
  * Side effects:
  *      Same as TimeSyncSet.
@@ -103,9 +106,19 @@ TimeSyncSet(Bool enable) // IN: status
 static int
 TimeSyncEnable(void)
 {
-   TimeSyncSet(TRUE);
-   ToolsCmd_Print("%s\n", SU_(option.enabled, "Enabled"));
-   return EXIT_SUCCESS;
+   int ret = EXIT_SUCCESS;
+   char *reply = NULL;
+   size_t replyLen;
+
+   if (TimeSyncSet(TRUE, &reply, &replyLen)) {
+      ToolsCmd_Print("%s\n", SU_(option.enabled, "Enabled"));
+   } else {
+      ToolsCmd_PrintErr(SU_(error.message, "Error: %s\n"), reply);
+      ret = EXIT_FAILURE;
+   }
+
+   vm_free(reply);
+   return ret;
 }
 
 
@@ -117,7 +130,7 @@ TimeSyncEnable(void)
  *      Disable time sync.
  *
  * Results:
- *      EXIT_SUCCESS
+ *      EXIT_SUCCESS on success, EXIT_FAILURE on failure.
  *
  * Side effects:
  *      Same as TimeSyncSet.
@@ -128,9 +141,19 @@ TimeSyncEnable(void)
 static int
 TimeSyncDisable(void)
 {
-   TimeSyncSet(FALSE);
-   ToolsCmd_Print("%s\n", SU_(option.disabled, "Disabled"));
-   return EXIT_SUCCESS;
+   int ret = EXIT_SUCCESS;
+   char *reply = NULL;
+   size_t replyLen;
+
+   if (TimeSyncSet(FALSE, &reply, &replyLen)) {
+      ToolsCmd_Print("%s\n", SU_(option.disabled, "Disabled"));
+   } else {
+      ToolsCmd_PrintErr(SU_(error.message, "Error: %s\n"), reply);
+      ret = EXIT_FAILURE;
+   }
+
+   vm_free(reply);
+   return ret;
 }
 
 

---